kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/Burrow.java

```diff
@@ -80,13 +80,26 @@ private HttpUrl address(String... paths) {
   }
 
   private Map<String, Object> request(String... paths) throws IOException {
+    Response response = null;
     HttpUrl url = address(paths);
     LOG.debug("GET {}", url);
     Request request = new Request.Builder()
         .url(url)
         .get().build();
-    Response response = client.newCall(request).execute();
-    return MAPPER.readValue(response.body().byteStream(), Map.class);
+    try {
+      response = client.newCall(request).execute();
+      // burrow will build a valid map with the body for invalid responses (i.e. consumer group not found), so we
+      // need to check that the response was failure.
+      if (!response.isSuccessful()) {
```

Comment:
is there an error code in the response that you want to include here?

Reply:
I was expecting that to be handled by the LOG message below with the response -  so we cover cases where the connection fails or doesn't json parse - rather than having two dumps of the response